index.js

```diff
@@ -771,6 +774,11 @@
     return x instanceof Right && R.eqProps('value', x, this);
   });
 
+  //  Right#extend :: Either a b ~> (b -> b) -> Either a b
```

Comment:
As discussed via HipChat, your interpretation is absolutely right, @pvrnav.

In `w.extend(f)`:

> `f` must return a value of type `v`, for some variable `v` contained in `w`.

So `f` must indeed be of type `v -> v` (or `b -> b` as written above).